app/controllers/orders_controller.rb

```diff
@@ -1,13 +1,24 @@
 class OrdersController < ApplicationController
+  before_action :authenticate_user!
+
   def add_to_order
-    if Order.last
-      order = Order.last
-    else
-      order = Order.create
+	  if session[:order_id] == nil || Order.find(session[:order_id]).finalized == true
```

Comment:
Well, that's a bold statement ;-) I would suggest to clear the `session[:order_id]` as well as setting the orders `finalized` attribute to true on checkout. That would take care of that problem.